schemas/extensions/schema/extension.schema.json

```diff
@@ -0,0 +1,197 @@
+          "$ref": "#/$defs/signal"
+        }
+      }
+    }
+  },
+  "properties": {
+    "schema": {
+      "title": "Extension Schema",
+      "description": "The schema version of the extension description syntax",
+      "default": "1-0-0",
+      "examples": [
+        "1-0-0",
+        "draft/2-0-0",
+        "2-1-0"
+      ],
+      "type": "string",
+      "pattern": "^(draft/)?[0-9]+-[0-9]+-[0-9]+$"
+    },
+    "name": {
+      "description": "The human-readable name of the extension.",
+      "examples": [
+        "Point Attractor",
+        "Interactive Marker",
+        "Weighted Sum"
+      ],
+      "type": "string"
+    },
+    "description": {
+      "description": "The human-readable description of this extension and its behaviour.",
+      "type": "object",
+      "additionalProperties": false,
+      "properties": {
+        "brief": {
+          "description": "A brief (one-line) description of the extension for quick reference and tool-tips.",
+          "examples": [
+            "Generates a velocity command towards a target attractor"
+          ],
+          "type": "string"
+        },
+        "details": {
+          "description": "An optional detailed description of the extension.",
+          "examples": [
+            "Given an input Cartesian pose and parameterized target pose, this component generates a Cartesian twist output towards the target linearly proportional to the distance multiplied by the gain parameter."
+          ],
+          "type": "string"
+        }
+      },
+      "required": [
+        "brief"
+      ]
+    },
+    "type": {
+      "title": "Extension Type",
+      "description": "The type of the extension.",
+      "type": "string",
+      "enum": [ "component", "lifecycle_component", "controller"]
+    },
+    "class": {
+      "$ref": "#/$defs/class"
+    },
+    "inherits": {
+      "$ref": "#/$defs/class"
+    },
+    "virtual": {
+      "title": "Virtual Extension",
+      "description": "If true, this property indicates that the extension is an abstract or virtual base class and not directly instantiable.",
+      "type": "boolean",
+      "default": false
+    },
+    "assignments": {
+      "title": "Assignments",
+      "description": "The assignments declared by the extension.",
+      "type": "object",
+      "patternProperties": {
+        "^[a-z]([a-z0-9_]?[a-z0-9])*$": {
+          "$ref": "#/$defs/assignment"
+        }
+      }
+    },
+    "inputs": {
+      "title": "Inputs",
+      "description": "The fixed input signals (subscriptions) of the extension.",
+      "$ref": "#/$defs/signals"
+    },
+    "outputs": {
+      "title": "Outputs",
+      "description": "The output signals (publications) of the extension.",
+      "$ref": "#/$defs/signals"
+    },
+    "parameters": {
+      "title": "Parameters",
+      "description": "The parameters declared by the extension.",
+      "type": "object",
+      "patternProperties": {
+        "^[a-z]([a-z0-9_]?[a-z0-9])*$": {
```

Comment:
I believe that this is too restrictive now. We should take it from here https://github.com/aica-technology/api/pull/260/files#diff-ea064547fa142cde57f15387faffd17eceda96b3fd8f8a55a8e536609c13d552

Reply:
Agreed, we can indeed be more permissive for parameter names and use `^[a-zA-Z0-9]([a-zA-Z0-9_.-]?[a-zA-Z0-9])*$`. Assignments will act as parameters and so they can share the same pattern.

Do you agree that for the rest (signals, predicates and services) it can stay more restrictive? For topics we should respect what the modulo SDK does, which coerces topic names to `[a-z0-9_]` but specifically requires starting with `[a-z]`, so pretty much what we have here. For predicates it's an open question but I would prefer to start more restrictive.

Reply:
the main thing is that those pattern properties (previously defined explicitly as e.g. `parameter_name`) are used with ROS interfaces to declare and set parameters and topic names. If the description metadata for a custom component declares that is has some parameter called `"&@(#! B!"`, it will fail at runtime because ROS will not allow that parameter name (either in the component constructor if the implementation has the same name, or else through the EventEngine that tries to set that parameter)

Reply:
Yeah we can't allow anything since ROS has it's own rules about admissible topic names. However, in modulo we are much stricter and we actually automatically remove characters from a string to define parameters, e.g. a user that does `add_parameter("te@ST", ....)` will end up with a parameter called `test`.

I do agree that for everything else we can leave the pattern properties as is since names of signals, predicates and service only match modulo components and no other open source components that we make a description for (+ topic names are actually that restrictive, where there are no restrictions for parameters in ROS at all i believe)